### PR TITLE
test: fixes for 32-bit archs

### PR DESCRIPTION
**Pull request by TinfoilSubmarine**
fixes https://github.com/prometheus/prometheus/issues/15547

### CODE REVIEW

File: model/histogram/test_utils.go [hunk 1]
```diff
@@ -19,12 +19,12 @@ func GenerateBigTestHistograms(numHistograms, numBuckets int) []*Histogram {
...
 	spanLength := uint32(bucketsPerSide / numSpans)
 	// Given all bucket deltas are 1, sum bucketsPerSide + 1.
-	observationCount := bucketsPerSide * (1 + bucketsPerSide)
+	observationCount := uint64(bucketsPerSide) * (1 + uint64(bucketsPerSide))
 
 	var histograms []*Histogram
 	for i := 0; i < numHistograms; i++ {
 		h := &Histogram{
-			Count:           uint64(i + observationCount),
+			Count:           uint64(uint64(i) + observationCount),
```
Comment (beorn7) [inline model/histogram/test_utils.go:27]: Th.e outer `uint64` isn't needed anymore.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thank you very much. See my comment. I hope that will fix the CI.

BTW: We want to address those glitches, see https://docs.google.com/document/d/1uurQCi5iVufhYHGlBZ8mJMK_freDFKPG0iYBQqJ9fvA/edit?pli=1&tab=t.0#bookmark=id.5ds2naruydu0

**Comment by beorn7** [comment]
You also need to [sign the DCO](https://github.com/prometheus/prometheus/pull/15548/checks?check_run_id=34054654996).

**Comment by TinfoilSubmarine** [comment]
The TestOOOHeadChunkReader_Chunk fix could probably be simplified, I just blindly swapped things to (u)int64 until there were no complaints and it fixed the test failure.

**Comment by TinfoilSubmarine** [comment]
Simplified the TestOOOHeadChunkReader_Chunk test fix a little bit.

**Review by beorn7 — APPROVED** [review]
Thank you very much.

**Comment by beorn7** [comment]
Special thanks for the clean commit layout.

**Pull request merged**